### PR TITLE
Navigation interface

### PR DESCRIPTION
**Pull request by Camonte**

### CODE REVIEW

File: app/src/main/java/ch/epfl/balelecbud/PointOfInterestActivity.java [hunk 1]
```diff
@@ -6,19 +6,25 @@
...
+import com.google.android.material.navigation.NavigationView;
+
 import ch.epfl.balelecbud.pointOfInterest.PointOfInterest;
 import ch.epfl.balelecbud.pointOfInterest.PointOfInterestData;
 import ch.epfl.balelecbud.pointOfInterest.PointOfInterestHolder;
 import ch.epfl.balelecbud.util.views.RecyclerViewData;
 import ch.epfl.balelecbud.util.views.RefreshableRecyclerViewAdapter;
 
-public class PointOfInterestActivity extends BasicActivity{
+public class PointOfInterestActivity extends BasicActivity implements NavigationView.OnNavigationItemSelectedListener{
```
Comment (Pialex99) [inline app/src/main/java/ch/epfl/balelecbud/PointOfInterestActivity.java:17]: `BasicActivity` already implements `NavigationView.OnNavigationItemSelectedListener`

File: app/src/main/java/ch/epfl/balelecbud/ScheduleActivity.java [hunk 2]
```diff
@@ -1,24 +1,35 @@
...
 
+import ch.epfl.balelecbud.friendship.SocialActivity;
 import ch.epfl.balelecbud.schedule.ScheduleAdapter;
 import ch.epfl.balelecbud.schedule.models.Slot;
 import ch.epfl.balelecbud.util.intents.FlowUtil;
 import ch.epfl.balelecbud.util.intents.IntentLauncher;
 
-public class ScheduleActivity extends BasicActivity {
-    private static final String TAG = ScheduleAdapter.class.getSimpleName();
+public class ScheduleActivity extends BasicActivity implements NavigationView.OnNavigationItemSelectedListener {
```
Comment (Pialex99) [inline app/src/main/java/ch/epfl/balelecbud/ScheduleActivity.java:27]: `BasicActivity` already implements `NavigationView.OnNavigationItemSelectedListener`

File: app/src/main/java/ch/epfl/balelecbud/BasicActivity.java [hunk 3]
```diff
@@ -1,18 +1,97 @@
...
+import com.google.android.material.navigation.NavigationView;
 
 import ch.epfl.balelecbud.authentication.Authenticator;
 import ch.epfl.balelecbud.authentication.FirebaseAuthenticator;
+import ch.epfl.balelecbud.friendship.SocialActivity;
 import ch.epfl.balelecbud.util.database.DatabaseWrapper;
 import ch.epfl.balelecbud.util.database.FirestoreDatabaseWrapper;
 
-public class BasicActivity extends AppCompatActivity {
+public class BasicActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {
```
Comment (Pialex99) [inline app/src/main/java/ch/epfl/balelecbud/BasicActivity.java:21]: Should be abstract since we should not instance it

Comment (axelmarmet) [reply app/src/main/java/ch/epfl/balelecbud/BasicActivity.java:21]: Yes you're right

File: app/src/main/java/ch/epfl/balelecbud/ScheduleActivity.java (hunk 2 above)
Comment (axelmarmet) [reply app/src/main/java/ch/epfl/balelecbud/ScheduleActivity.java:27]: Yup, missed it during the refactoring

File: app/src/main/java/ch/epfl/balelecbud/PointOfInterestActivity.java (hunk 1 above)
Comment (axelmarmet) [reply app/src/main/java/ch/epfl/balelecbud/PointOfInterestActivity.java:17]: Yup, missed it during the refactoring

**Pull request merged**